metacat-common/src/main/java/com/netflix/metacat/common/QualifiedName.java

```diff
@@ -550,33 +593,88 @@ public String toString() {
         return qualifiedName;
     }
 
+    /**
+     *
+     * Checks if a CharSequence is empty ("") or null.
+     */
+    private static boolean isNullOrEmpty(@Nullable final CharSequence cs) {
+        return cs == null || cs.length() == 0;
+    }
+
     /**
      * Type of the connector resource.
      */
     public enum Type {
         /**
          * Catalog type.
          */
-        CATALOG,
+        CATALOG("CATALOG"),
 
         /**
          * Database type.
          */
-        DATABASE,
+        DATABASE("DATABASE"),
 
         /**
          * Table type.
          */
-        TABLE,
+        TABLE("TABLE"),
 
         /**
          * Partition type.
          */
-        PARTITION,
+        PARTITION("PARTITION"),
 
         /**
          * MView type.
          */
-        MVIEW
+        MVIEW("MVIEW");
+
+
+        /**
+         * TYPE REGEX MAP.
+         */
+        private static final Map<Type, String>
```

Comment:
Instead of using the Map, set the mysqlRegexValue during initialization of an enum. Ex: CATALOG(""^([^\\/]+)$")

Reply:
This looks odd to me, binding the regex into individual enum value.

Reply:
We are binding any context we have about the enum to it. Having it in the Map in this class does not make sense.

Reply:
This regext matching helper doesn't belong to this class, I agree. We can bind to the field if you have a strong opinion, even though I feel it's very strange.